tests/core/test_local_service.py

```diff
@@ -12,38 +12,328 @@
+    assert isinstance(local_service.metadata_stores, dict)
+    assert isinstance(
+        local_service.metadata_stores[LOCAL_METADATA_STORE_NAME],
+        BaseMetadataStore,
+    )
+
+
+def test_local_service_can_access_artifact_stores(tmp_path: str) -> None:
+    """Local Service can access artifact stores."""
+    Repo.init(tmp_path)
+    repo = Repository(str(tmp_path))
+    local_service = repo.get_service()
+    assert local_service.artifact_stores is not None
+    assert isinstance(local_service.artifact_stores, dict)
+    assert isinstance(
+        local_service.artifact_stores[LOCAL_ARTIFACT_STORE_NAME],
+        BaseArtifactStore,
+    )
+
+
+def test_local_service_can_access_orchestrators(tmp_path: str) -> None:
+    """Local Service can access orchestrators."""
+    Repo.init(tmp_path)
+    repo = Repository(str(tmp_path))
+    local_service = repo.get_service()
+    assert local_service.orchestrators is not None
+    assert isinstance(local_service.orchestrators, dict)
+    assert isinstance(
+        local_service.orchestrators[LOCAL_ORCHESTRATOR_NAME],
+        BaseOrchestrator,
+    )
+
+
+def test_get_stack_returns_a_stack_when_provided_a_key(tmp_path: str) -> None:
+    """Check get_stack returns a stack with expected properties."""
+    Repo.init(tmp_path)
+    repo = Repository(str(tmp_path))
+    local_service = repo.get_service()
+    stack = local_service.get_stack("local_stack")
+    assert stack is not None
+    assert isinstance(stack, BaseStack)
+    assert (
+        stack.orchestrator
+        == local_service.orchestrators[LOCAL_ORCHESTRATOR_NAME]
+    )
+    assert (
+        stack.artifact_store
+        == local_service.artifact_stores[LOCAL_ARTIFACT_STORE_NAME]
+    )
+    assert (
+        stack.metadata_store
+        == local_service.metadata_stores[LOCAL_METADATA_STORE_NAME]
+    )
+
+
+def test_register_stack_works_as_expected(tmp_path: str) -> None:
+    """Test register_stack method registers a stack as expected."""
+    Repo.init(tmp_path)
+    repo = Repository(str(tmp_path))
+    local_service = repo.get_service()
+    local_stack1 = local_service.get_stack("local_stack")
+    local_service.register_stack("local_stack_2", local_stack1)
+    local_stack2 = local_service.get_stack("local_stack_2")
+    assert local_stack2 is not None
+    assert local_stack2.orchestrator == local_stack1.orchestrator
+    assert local_stack2.artifact_store == local_stack1.artifact_store
+    assert local_stack2.metadata_store == local_stack1.metadata_store
+    # TODO: [Medium] rework this as a fixture to be run after the test completes
+    local_service.delete_stack("local_stack_2")
+
+
+def test_get_stack_raises_exception_when_key_does_not_exist(
+    tmp_path: str,
+) -> None:
+    """Test get_stack raises exception when key does not exist."""
+    Repo.init(tmp_path)
+    repo = Repository(str(tmp_path))
+    local_service = repo.get_service()
+    with pytest.raises(Exception):
```

Comment:
I think we should specify which exception is raised here and in some of the tests below

Reply:
What do you think about checking for a `DoesNotExistException` exception here and in similar tests? In this case we're not using any CLI runner so should be possible to simply change the exception

Reply:
I changed all the exceptions to their specific exceptions rather than the generic.